src/rules/declaration-block-no-duplicate-properties/index.js

```diff
@@ -61,6 +67,31 @@ export default function (on, options) {
         const indexDuplicate = decls.indexOf(prop.toLowerCase())
 
         if (indexDuplicate !== -1) {
+
+          if (optionsHaveIgnored(options, "consecutive-duplicates-with-different-values")) {
+            // if duplicates are not consecutive
+            if (indexDuplicate !== decls.length - 1) {
+              report({
+                message: messages.rejected(prop),
+                node: child,
+                result,
+                ruleName,
+              })
+              return
+            }
+            // if values of consecutive duplicates are equal
+            if (value === values[indexDuplicate]) {
+              report({
+                message: messages.rejected(value),
+                node: child,
+                result,
+                ruleName,
+              })
+              return
+            }
+            return
+          }
+
           if (
             optionsHaveIgnored(options, "consecutive-duplicates")
```

Comment:
This line, along with the imports should be fixed if you rebase against stylelint/master.